modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java

```diff
@@ -368,67 +444,217 @@ private void _testSystemObjectRelationshipManyToMany() throws Exception {
 				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
 				StringUtil.randomId(),
 				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
 
 		String objectFieldNamePrefix =
-			"r_" + objectRelationship.getName() + "_";
+			"r_" + objectRelationship1.getName() + "_";
 
 		Assert.assertTrue(
 			_hasColumn(
-				_systemObjectDefinition.getExtensionDBTableName(),
+				_systemObjectDefinition2.getExtensionDBTableName(),
 				objectFieldNamePrefix +
 					_objectDefinition1.getPKObjectFieldName()));
 		Assert.assertNotNull(
 			_objectFieldLocalService.fetchObjectField(
-				_systemObjectDefinition.getObjectDefinitionId(),
+				_systemObjectDefinition2.getObjectDefinitionId(),
 				objectFieldNamePrefix +
 					_objectDefinition1.getPKObjectFieldName()));
 
 		_objectRelationshipLocalService.deleteObjectRelationship(
-			objectRelationship);
+			objectRelationship1);
 
 		Assert.assertNull(
 			_objectFieldLocalService.fetchObjectField(
-				_systemObjectDefinition.getObjectDefinitionId(),
+				_systemObjectDefinition2.getObjectDefinitionId(),
 				objectFieldNamePrefix +
 					_objectDefinition1.getPKObjectFieldName()));
+
+		try {
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				_systemObjectDefinition1.getObjectDefinitionId(),
+				_objectDefinition1.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				StringUtil.randomId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+		}
+		catch (ObjectRelationshipParameterObjectFieldIdException
+					objectRelationshipParameterObjectFieldIdException) {
+
+			Assert.assertEquals(
+				"Object definition " + _systemObjectDefinition1.getName() +
+					" requires a parameter object field ID",
+				objectRelationshipParameterObjectFieldIdException.getMessage());
+		}
+
+		long randomParameterObjectFieldId = RandomTestUtil.randomLong();
+
+		try {
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				_systemObjectDefinition1.getObjectDefinitionId(),
+				_objectDefinition1.getObjectDefinitionId(),
+				randomParameterObjectFieldId,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				StringUtil.randomId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+		}
+		catch (ObjectRelationshipParameterObjectFieldIdException
+					objectRelationshipParameterObjectFieldIdException) {
+
+			Assert.assertEquals(
+				"Parameter object field ID " + randomParameterObjectFieldId +
+					" does not exist",
+				objectRelationshipParameterObjectFieldIdException.getMessage());
+		}
+
+		List<ObjectField> objectDefinition2ObjectFields =
+			_objectFieldLocalService.getObjectFields(
+				_objectDefinition2.getObjectDefinitionId());
+
+		ObjectField objectDefinition2ObjectField =
+			objectDefinition2ObjectFields.get(0);
+
```

Comment:
Not sure about these names. I think we could just call them  `objectFields` and `objectField`.

Reply:
Sure thing. I'll make this change